File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
  let stack = [];
  
    //todo 괄호 닫는 문제는 stack
    //여는 표시만 넣으면됨
    for(let str of s){
        let top = stack[stack.length -1];
        if(!!top && top == "[" && str == "]"){
            stack.pop();
        } else if (!!top && top == "{" && str == "}"){
            stack.pop();
        } else if (!! top && top == "(" && str == ")"){
            stack.pop();
        } else {
            stack.push(str)
        }
    }

    return stack.length > 0 ? false : true;
};

File: 0020-valid-parentheses/0020-valid-parentheses.go
func isValid(s string) bool {
    items := make([]rune,0)

    //스택에 담기

    for _, value := range s {
        //만약에 
        if value == '(' || value == '{' || value == '['  {
            //이라면 , 
            items = append(items, value)
        } else {
            if len(items) == 0 {return false}
            top := items[len(items) -1]
            //top 이 
            if top == '(' && value == ')' {
                items = items[:len(items) -1] // 마지막 요소 제거
            } else if top == '{' && value == '}' {
                items = items[:len(items)-1]
            } else if top == '[' && value == ']' {
                items = items[:len(items)-1]
            } else {
                return false;
            }

        }
    }

    return len(items) == 0
}